src/h5web/visualizations/line/ErrorBarCurve.tsx

```diff
@@ -1,51 +1,73 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useMemo } from 'react';
 import { LineSegments } from 'react-three-fiber/components';
-import { useDataGeometry } from './hooks';
+import { BufferGeometry } from 'three';
+import { lineSegment } from '../shared/utils';
+import { useCanvasScales } from '../shared/hooks';
 
 const DEFAULT_COLOR = '#1b998b';
 
+const CAP_LENGTH = 10;
+
 interface Props {
   abscissas: number[];
   ordinates: number[];
   errors: number[];
   color?: string;
+  capLength?: number;
 }
 
 function ErrorBarCurve(props: Props): ReactElement {
-  const { abscissas, ordinates, errors, color = DEFAULT_COLOR } = props;
-
-  const capLength = 0.1 * (abscissas[1] - abscissas[0]);
-
-  const errorAbscissas = [];
-  const errorOrdinates = [];
-  for (const [index, x] of abscissas.entries()) {
-    const y = ordinates[index];
-    const yerr = errors[index];
-
-    if (yerr === 0) {
-      continue;
-    }
-
-    // First segment: bottom cap
-    errorAbscissas.push(x - capLength);
-    errorOrdinates.push(y - yerr);
-    errorAbscissas.push(x + capLength);
-    errorOrdinates.push(y - yerr);
-
-    // Second segment: error bar
-    errorAbscissas.push(x);
-    errorOrdinates.push(y - yerr);
-    errorAbscissas.push(x);
-    errorOrdinates.push(y + yerr);
-
-    // Third segment: top cap
-    errorAbscissas.push(x - capLength);
-    errorOrdinates.push(y + yerr);
-    errorAbscissas.push(x + capLength);
-    errorOrdinates.push(y + yerr);
-  }
-
-  const errorGeometry = useDataGeometry(errorAbscissas, errorOrdinates);
+  const {
+    abscissas,
+    ordinates,
+    errors,
+    color = DEFAULT_COLOR,
+    capLength = CAP_LENGTH,
+  } = props;
+
+  const { abscissaScale, ordinateScale } = useCanvasScales();
+
+  const errorGeometry = useMemo(() => {
+    const points = ordinates.flatMap((val, index) => {
+      const [x, y] = [abscissaScale(abscissas[index]), ordinateScale(val)];
+
+      if (!Number.isFinite(y)) {
+        return [];
+      }
+
+      const error = errors[index];
+      if (error === 0) {
+        return [];
+      }
+      const yerr_min = ordinateScale(val - error);
+      const yerr_max = ordinateScale(val + error);
+
+      return [
+        // First segment: bottom cap
+        ...lineSegment(
+          x - capLength / 2,
+          yerr_min,
+          x + capLength / 2,
+          yerr_min
+        ),
+        // Second segment: bottom error bar
+        ...lineSegment(x, yerr_min, x, y),
+        // Third segment: top error bar
+        ...lineSegment(x, y, x, yerr_max),
```

Comment:
The error bar is now in two segments to be able to show the bottom/top part even though the other is not shown.